Development/nmos/streamcompatibility_api.cpp

```diff
@@ -0,0 +1,907 @@
+                            {
+                                set_error_reply(res, status_codes::BadRequest, U("The requested Constraint Set uses Parameter Constraints unsupported by this Sender."));
+                            }
+                        }
+                        else
+                        {
+                            slog::log<slog::severities::warning>(gate, SLOG_FLF) << "Active Constraints update is requested for " << id_type << " but this operation is locked";
+                            set_error_reply(res, status_codes::Locked);
+                        }
+                    }
+                    else if (nmos::details::is_erased_resource(resources, id_type))
+                    {
+                        set_error_reply(res, status_codes::NotFound, U("Not Found; ") + nmos::details::make_erased_resource_error());
+                    }
+                    else
+                    {
+                        set_reply(res, status_codes::NotFound);
+                    }
+
+                    return true;
+                });
+            });
+
+            streamcompatibility_api.support(U("/") + nmos::patterns::senderType.pattern + U("/") + nmos::patterns::resourceId.pattern + U("/constraints/active/?"), methods::DEL, [&model, effective_edid_setter, active_constraints_handler, &gate_](http_request req, http_response res, const string_t&, const route_parameters& parameters)
+            {
+                nmos::api_gate gate(gate_, req, parameters);
+                auto lock = model.write_lock();
+                auto& resources = model.streamcompatibility_resources;
+
+                const string_t resourceId = parameters.at(nmos::patterns::resourceId.name);
+
+                const std::pair<nmos::id, nmos::type> id_type{ resourceId, nmos::types::sender };
+                const auto streamcompatibility_sender = find_resource(resources, id_type);
+
+                if (resources.end() != streamcompatibility_sender)
+                {
+                    const auto& endpoint_active_constraints = nmos::fields::endpoint_active_constraints(streamcompatibility_sender->data);
+
+                    if (!nmos::fields::temporarily_locked(endpoint_active_constraints))
+                    {
+                        slog::log<slog::severities::info>(gate, SLOG_FLF) << "Active Constraints deletion is requested for " << id_type;
+
+                        const auto active_constraints = web::json::value_of({
+                            { nmos::fields::constraint_sets, web::json::value::array() }
+                        });
+                        web::json::value intersection = web::json::value::array();
+
+                        active_constraints_handler(*streamcompatibility_sender, active_constraints, intersection);
+                        details::set_active_constraints(model, resourceId, nmos::fields::constraint_sets(active_constraints), intersection, effective_edid_setter);
```

Comment:
Shouldn't `active_constraints_handler` be checked for null before use? If it is null, should it return 500?

Reply:
Or `active_constraints_handler` is not actually required, as we just want to clear the `intersection_of_caps_and_constraints` and the `endpoint_active_constraints`